### PR TITLE
Add Outbox validation

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

File: src/NServiceBus.AzureFunctions.InProcess.ServiceBus/Serverless/OutboxProcessingValidationBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    public OutboxProcessingValidationBehavior(IReadOnlySettings settings)
+    {
+        outboxEnabled = settings.IsFeatureActive(typeof(Outbox));
+    }
+
+    public Task Invoke(ITransportReceiveContext context, Func<ITransportReceiveContext, Task> next)
+    {
+        var state = context.Extensions.Get<State>();
+        if (outboxEnabled && state.IsAtomic)
```
Comment (andreasohlund) [inline src/NServiceBus.AzureFunctions.InProcess.ServiceBus/Serverless/OutboxProcessingValidationBehavior.cs:21]: We could also check for the presence of the commitable transaction but perhaps this is more explicit?

Comment (SzymonPobiega) [reply src/NServiceBus.AzureFunctions.InProcess.ServiceBus/Serverless/OutboxProcessingValidationBehavior.cs:21]: Yeah, unless there is a significant performance impact of having that state object, I prefer the explicit version.

**Review by WojcikMike — APPROVED** [review]

File: src/ServiceBus.AcceptanceTests/When_using_processatomic_with_outbox.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            public FunctionHandler(TransportTransactionMode transactionMode) : base(transactionMode)
+            {
+                CustomizeConfiguration = configuration =>
+                {
+                    configuration.AdvancedConfiguration.Pipeline.Register(b => new FailBeforeAckBehavior(b.GetRequiredService<Context>()),
+                        "Simulates a failure in ACKing the incoming message");
+                    configuration.AdvancedConfiguration.EnableOutbox();
+                    configuration.AdvancedConfiguration.UsePersistence<NonDurablePersistence>();
+                    //configuration.AdvancedConfiguration.Recoverability().Immediate(x => x.NumberOfRetries(1));
```
Comment (andreasohlund) [inline src/ServiceBus.AcceptanceTests/When_using_processatomic_with_outbox.cs:46]: Remove

**Review by andreasohlund — APPROVED** [review]

**Review by andreasohlund — CHANGES_REQUESTED** [review]

**Review by andreasohlund — APPROVED** [review]

**Comment by andreasohlund** [comment]
@SzymonPobiega you need to adjust the test to the new ex message

**Pull request merged**